exec.go

```diff
@@ -493,27 +493,27 @@ func local(et, _, argt *Text, _, _ bool, arg string) {
 // putfile writes File to disk, if it's safe to do so.
 //
 // TODO(flux): Write this in terms of the various cases.
-func putfile(f *File, q0 int, q1 int, name string) error {
-	w := f.curtext.w
+func putfile(oeb *ObservableEditableBuffer, q0 int, q1 int, name string) error {
+	w := oeb.GetCurObserver().(*Text).w
 	d, err := os.Stat(name)
 
 	// Putting to the same file that we already read from.
-	if err == nil && name == f.name {
-		if !os.SameFile(f.info, d) || d.ModTime().Sub(f.info.ModTime()) > time.Millisecond {
-			f.UpdateInfo(name, d)
+	if err == nil && name == oeb.Name() {
+		if !os.SameFile(oeb.f.details.Info, d) || d.ModTime().Sub(oeb.f.details.Info.ModTime()) > time.Millisecond {
```

Comment:
Some kind of improved encapsulation is needed here. Conceivably that can happen in a later CL when you relocate`details` into `ObservableEditableBuffer`.

Reply:
I already have that CL compleated, just not up to date with it's parent (this branch) I'll have to see what I did there